internal/topic/topic-operation.go

```diff
@@ -111,6 +131,34 @@ func (operation *Operation) CreateTopics(topics []string, flags CreateTopicFlags
 		topicDetails.ConfigEntries[configParts[0]] = &configParts[1]
 	}
 
+	if flags.File != "" {
+		fileContent, err := os.ReadFile(flags.File)
+		if err != nil {
+			return errors.Wrap(err, "could not read topic description file")
+		}
+
+		createTopicConfig := CreateTopicConfig{}
+		ext := path.Ext(flags.File)
+		var unmarshalErr error
+		switch ext {
+		case ".yml", ".yaml":
+			unmarshalErr = yaml.Unmarshal(fileContent, &createTopicConfig)
+		case ".json":
+			unmarshalErr = json.Unmarshal(fileContent, &createTopicConfig)
+		default:
+			return errors.Wrapf(err, "unsupported file format '%s'", ext)
+		}
+		if unmarshalErr != nil {
+			return errors.Wrap(err, "could not umarshal config file")
+		}
+
+		topicDetails.NumPartitions = int32(len(createTopicConfig.Partitions))
+		topicDetails.ReplicationFactor = int16(len(createTopicConfig.Partitions[0].Replicas))
```

Comment:
I am not super sure about the ReplicationFactor... 
Maybe if it cannot be reliably determined, it should be removed.